python/lsst/ap/pipe/createApFakes.py

```diff
@@ -144,129 +147,55 @@ def run(self, tractId, skyMap):
             Catalog of random points covering the given tract. Follows the
             columns and format expected in `lsst.pipe.tasks.InsertFakes`.
         """
-        # Use the tractId as the ranomd seed.
+        # Use the tractId as the random seed.
         rng = np.random.default_rng(tractId)
-        tractBoundingCircle = \
-            skyMap.generateTract(tractId).getOuterSkyPolygon().getBoundingCircle()
-        tractArea = tractBoundingCircle.getArea() * (180 / np.pi) ** 2
-        nFakes = int(self.config.fakeDensity * tractArea)
+
+        tract = skyMap.generateTract(tractId)
+        tractWcs = tract.getWcs()
+        vertexList = tract.getVertexList()
+        vertexRas = [vertex.getRa().asDegrees() for vertex in vertexList]
+        vertexDecs = [vertex.getDec().asDegrees() for vertex in vertexList]
+
+        catalog = generate_injection_catalog(
+            ra_lim=sorted([np.min(vertexRas), np.max(vertexRas)]),
+            dec_lim=sorted([np.min(vertexDecs), np.max(vertexDecs)]),
+            density=self.config.fakeDensity,
+            source_type="Star",
+            seed=str(tractId),
+            wcs=tractWcs
+        )
+
+        nFakes = len(catalog)
 
         self.log.info(
             f"Creating {nFakes} star fakes over tractId={tractId} with "
-            f"bounding circle area: {tractArea:.4f} deg^2 and "
```

Comment:
A log message with some info on the area covered would still be helpful. Perhaps log the min/max RA and Dec from `vertexRas` and `vertexDecs`.

Reply:
I added the area and coordinate range. Although I think we should mark this for deprecation soon enough.